src/log.h

```diff
@@ -119,9 +121,8 @@ class StreamLogOutput : public ICombinedLogOutput {
 
 	void logRaw(LogLevel lev, const std::string &line)
 	{
-		static const std::string use_logcolor = g_settings->get("log_color");
-
-		bool colored = use_logcolor == "detect" ? is_tty : use_logcolor == "yes";
+		bool colored = (Logger::use_logcolor == "auto") ? is_tty :
```

Comment:
ouch, and ugly condition on each log. Don't use a string but a enum on u8 instead

Reply:
A simple boolean should be sufficient, is_tty is set only once for each stream after all.